### PR TITLE
chore: final review fixes

### PR DESCRIPTION
**Pull request by howydev**
1. Move all global "using" statements to the top of the library file
2. Remove redundant bit masking operations being performed for some library helper functions
3. Removed redundant variable and flag in a library helper function

### CODE REVIEW

**Review by jaypaik — APPROVED** [review]

**Comment by howydev** [comment]
updated tests to check vals with dirty upper bits also, to showcase that we can remove the masking for  `uint32 entityId`. for `address directCallValidation` since the type has size `bytes20` and `ValidationLookupKey` is size `bytes21`, the rightshift should be sufficient

File: src/libraries/ValidationLocatorLib.sol [hunk 1]
```diff
@@ -176,14 +179,14 @@ library ValidationLocatorLib {
     // Only safe to call if the lookup has been asserted to be a direct call validation.
     function directCallAddress(ValidationLookupKey _lookupKey) internal pure returns (address result) {
         assembly ("memory-safe") {
-            result := and(shr(8, _lookupKey), 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF)
+            result := shr(8, _lookupKey)
```
Comment (jaypaik) [inline src/libraries/ValidationLocatorLib.sol]: @adamegyed Thoughts on these changes?

**Review by adamegyed — APPROVED** [review]
Looks good!

File: src/libraries/ValidationLocatorLib.sol [hunk 2]
```diff
@@ -183,7 +186,7 @@ library ValidationLocatorLib {
     // Only safe to call if the lookup has been asserted to be a non-direct call validation.
     function entityId(ValidationLookupKey _lookupKey) internal pure returns (uint32 result) {
         assembly ("memory-safe") {
-            result := and(shr(8, _lookupKey), 0xFFFFFFFFFFFFFFFF)
+            result := and(shr(8, _lookupKey), 0xFFFFFFFF)
```
Comment (jaypaik) [inline src/libraries/ValidationLocatorLib.sol:189]: Was this change intentional?

Comment (howydev) [reply src/libraries/ValidationLocatorLib.sol:189]: yep - uint32 is only 4 bytes long, our prev mask is 8 bytes. surfaced by QS: https://github.com/alchemyplatform/modular-account/pull/293#discussion_r1860023609
